runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -257,14 +262,54 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		{
 			type -= MENU_ACTION_DEPRIORITIZE_OFFSET;
 		}
-
-		if (config.highlightMenuNames() &&
-			NPC_MENU_ACTIONS.contains(MenuAction.of(type)) &&
-			highlightedNpcs.stream().anyMatch(npc -> npc.getIndex() == event.getIdentifier()))
+		if (NPC_MENU_ACTIONS.contains(MenuAction.of(type)))
 		{
 			MenuEntry[] menuEntries = client.getMenuEntries();
 			final MenuEntry menuEntry = menuEntries[menuEntries.length - 1];
-			final String target = ColorUtil.prependColorTag(Text.removeTags(event.getTarget()), config.getHighlightColor());
+
+			Matcher matcher = COLOR_REGEX.matcher(menuEntry.getTarget());
```

Comment:
All of this color selection code should be moved to a separate method to keep the menu entry added subscriber's body clean and straightforward.